### DISCUSSION TITLE
Pictures, Exif and tags

### DISCUSSION THREAD

**Discussion by MrCasquette** [directus/directus#16414] [Archive]
Is there anyway to access exif or some tags in files in directus. I use Lightroom to list my pictures with some people faces, and want to add it in the file tag. So i can access it and sort files by people.
If not it could be an idea. Thanks for your time :)

**Comment by rijkvanzanten** [comment]
Exif is already scraped and stored under `directus_files.metadata`, so as soon as #15889 is merged, this should be possible 🙂

**Reply by MrCasquette** [reply]
You are genius. Thanks for the reply. 😊

**Reply by MrCasquette** [reply]
I don't have news about #15889 and I want to know if it will be coming soon. And to be sure, this will be add more field into metadata like tags for example, so i can access some tags i get on my pictures? Thanks for your time @rijkvanzanten we are lucky to have you here on earth :D You helped me so many times directly and by reading your answers on people discussion.

**Reply by paescuj** [reply]
> I don't have news about https://github.com/directus/directus/pull/15889 and I want to know if it will be coming soon.

See https://github.com/directus/directus/pull/15889#issuecomment-1308510035

 > And to be sure, this will be add more field into metadata like tags for example, so i can access some tags i get on my pictures?

As @rijkvanzanten mentioned the metadata is already stored (including tags under `metadata.ifd0`) so how do you mean this? Are you referring to the filter in the UI?

**Reply by MrCasquette** [reply]
Ok thanks for the reply. I try but get nothing in the metadata when i upload a file with tag. Like firstname for examples. I only get informations like camera model, aperture etc... Maybe it need an additional step that i've missed. Could be more precise about ifd0 what does it means?

**Reply by paescuj** [reply]
I suppose you're talking about the sidebar in the UI? @rijkvanzanten and me were talking about the data retrieved via the API, see https://docs.directus.io/reference/files.html#the-file-object. In the API response you have the `metadata` object and inside this object the tags are stored within the `ifd0` field.

**Reply by MrCasquette** [reply]
I was talking about both. I tried the API too and couldn't find anything. Maybe the problem is with my import? I'll look a bit more and hope to find

**Reply by paescuj** [reply]
Okay, I see. I've tested it myself by uploading a JPEG file and the tags were properly listed under `metadata.ifd0`...

**Reply by MrCasquette** [reply]
Maybe it not shown in my console.log but exist. I will try it soon. I Really thank you for your time

**Reply by paescuj** [reply]
Hmm, maybe you need to use something like `console.log(JSON.stringify(myApiResponse, null, 4))` to print the full object, see https://stackoverflow.com/a/10729391/20504834. You're welcome!

**Reply by MrCasquette** [reply]
Ok just tried and unfortunately it doesn't work. Indeed metadata.ifd0 appears (i forget it but i already seen it)
https://ibb.co/VjNKGWb
But as I said no tags appears while on my file system (macos) it does. As you can see
https://ibb.co/WW0R9T2
"Mots-Clés" at the first line is Tag in french
An idea ? 😅

**Reply by paescuj** [reply]
Ah yes, you need to explicitly allow this metadata via the environment variable `FILE_METADATA_ALLOW_LIST`, like this:
```
FILE_METADATA_ALLOW_LIST=ifd0.Make,ifd0.Model,exif.FNumber,exif.ExposureTime,exif.FocalLength,exif.ISO,iptc.Keywords
```

The keywords will then appear under "iptc".

See https://docs.directus.io/self-hosted/config-options.html#metadata for more information.

**Reply by MrCasquette** [reply]
You are awesome, I even had never seen this option on the .env !!!!!!!
It works !!!!!!!! Oh I'm so happy to work with this Thank you, thank you directus team for being awesome

**Reply by paescuj** [reply]
Thank you for the kind words! ❤️  I'm glad it's working 😃
